src/fmc.rs

```diff
@@ -0,0 +1,564 @@
+use crate::gpio::gpiob::{PB5, PB6};
+use crate::gpio::gpioc::{PC0, PC2, PC3};
+use crate::gpio::gpioh::{PH2, PH3, PH5, PH6, PH7};
+use crate::gpio::GpioExt;
+use crate::stm32::FMC;
+
+pub use crate::stm32::fmc::{
+    sdcmr::MODE_AW,
+    sdcr::{CAS_A, MWID_A, NB_A, NC_A, NR_A, RPIPE_A, SDCLK_A},
+    SDCMR, SDCR, SDRTR, SDSR, SDTR,
+};
+
+type FMCAF = crate::gpio::Alternate<crate::gpio::AF12>;
+macro_rules! fmc_gpio {
+    ($gpioX:ident.$pXY:ident) => {
+        $gpioX
+            .$pXY
+            .into_alternate_af12()
+            .set_speed(crate::gpio::Speed::High)
```

Comment:
Why only `High` instead of `VeryHigh`?

Reply:
Primarily because it's unnecessary. Using a higher `Speed` means using more GPIO drivers (more drive strength), which means more potential EMI emissions and power consumption (best practice is to use the lowest Speed suitable for the application). The FMC runs the memory at half of HCLK, which puts you in the ballpark of 90MHz `SDCLK` frequency on a maxed out F429 running at 180MHz. Speed::High the lowest setting which gives us the bandwidth we need.